snapcraft/internal/pluginhandler.py

```diff
@@ -417,13 +417,28 @@ def clean_build(self, hint=''):
 
     def migratable_fileset_for(self, step):
         plugin_fileset = self.code.snap_fileset()
-        fileset = (getattr(self.code.options, step, ['*']) or ['*']).copy()
+        if step == 'prime':
+            step = 'snap'
+        fileset = self._get_fileset(step).copy()
+        # If we're priming and we don't have an explicit set of files to prime
+        # include the files from the stage step
+        if step == 'prime' and fileset == ['*']:
```

Comment:
In case you don't see the comment on the other branch, This is a condition that will never be True.
if step is 'prime', the if above will change it to 'snap'.